library/core/src/num/mod.rs

```diff
@@ -1032,29 +1028,52 @@ fn from_str_radix<T: FromStrRadixHelper>(src: &str, radix: u32) -> Result<T, Par
     // of multi-byte sequences
     let src = src.as_bytes();
 
-    let (is_positive, digits) = match src[0] {
-        b'+' | b'-' if src[1..].is_empty() => {
-            return Err(PIE { kind: InvalidDigit });
+    let (first, mut digits) = (*src.get(0).ok_or_else(|| PIE { kind: Empty })?, &src[1..]);
+
+    let (is_positive, mut result) = match first {
```

Comment:
It looks like there's a bit more repetition going on here than there needs to be.

Maybe try this with slice patterns, or something?  I'm imagining something like
```rust
let (is_positive, digits) = match src {
    [b'-', d] => (false, d),
    [b'+', d] => (true, d),
    d => (true, d),
};
```
To hopefully simplify a bit of the `first`/`digits`/`result` dance that's currently happening.